pkg/controller/gitopsservice/gitopsservice_controller.go

```diff
@@ -287,168 +307,191 @@ func (r *ReconcileGitopsService) Reconcile(request reconcile.Request) (reconcile
-	serviceAccountObj := newServiceAccount(serviceNamespacedName)
+	{
+		serviceAccountObj := newServiceAccount(gitopsserviceNamespacedName)
 
-	// Set GitopsService instance as the owner and controller
-	if err := controllerutil.SetControllerReference(instance, serviceAccountObj, r.scheme); err != nil {
-		return reconcile.Result{}, err
-	}
+		// Set GitopsService instance as the owner and controller
+		if err := controllerutil.SetControllerReference(instance, serviceAccountObj, r.scheme); err != nil {
+			return reconcile.Result{}, err
+		}
 
-	existingServiceAccount := &corev1.ServiceAccount{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: serviceAccountObj.Namespace, Name: serviceAccountObj.Name}, existingServiceAccount)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			reqLogger.Info("Creating a new ServiceAccount", "Namespace", serviceAccountObj.Namespace, "Name", serviceAccountObj.Name)
-			err = r.client.Create(context.TODO(), serviceAccountObj)
-			if err != nil {
+		existingServiceAccount := &corev1.ServiceAccount{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: serviceAccountObj.Namespace, Name: serviceAccountObj.Name}, existingServiceAccount)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				reqLogger.Info("Creating a new ServiceAccount", "Namespace", serviceAccountObj.Namespace, "Name", serviceAccountObj.Name)
+				err = r.client.Create(context.TODO(), serviceAccountObj)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+			} else {
 				return reconcile.Result{}, err
 			}
-		} else {
-			return reconcile.Result{}, err
 		}
 	}
 
 	// Define a new cluster role for backend service
-	clusterRoleObj := newClusterRole(serviceNamespacedName)
+	{
+		clusterRoleObj := newClusterRole(gitopsserviceNamespacedName)
 
-	// Set GitopsService instance as the owner and controller
-	if err := controllerutil.SetControllerReference(instance, clusterRoleObj, r.scheme); err != nil {
-		return reconcile.Result{}, err
-	}
+		// Set GitopsService instance as the owner and controller
+		if err := controllerutil.SetControllerReference(instance, clusterRoleObj, r.scheme); err != nil {
+			return reconcile.Result{}, err
+		}
 
-	existingClusterRole := &rbacv1.ClusterRole{}
-	err = r.client.Get(context.TODO(), types.NamespacedName{Name: clusterRoleObj.Name}, existingClusterRole)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			reqLogger.Info("Creating a new Cluster Role", "Name", clusterRoleObj.Name)
-			err = r.client.Create(context.TODO(), clusterRoleObj)
+		existingClusterRole := &rbacv1.ClusterRole{}
+		err := r.client.Get(context.TODO(), types.NamespacedName{Name: clusterRoleObj.Name}, existingClusterRole)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				reqLogger.Info("Creating a new Cluster Role", "Name", clusterRoleObj.Name)
+				err = r.client.Create(context.TODO(), clusterRoleObj)
+				if err != nil {
+					return reconcile.Result{}, err
```

Comment:
FYI I made no change to this block of besides whitespace, not sure why GitHub is considering it separate...